libraries/mbed/targets/hal/TARGET_NXP/TARGET_LPC82X/pwmout_api.c

```diff
@@ -147,11 +155,17 @@ void pwmout_period_ms(pwmout_t* obj, int ms)
 // Set the PWM period, keeping the duty cycle the same.
 void pwmout_period_us(pwmout_t* obj, int us)
 {
-    uint32_t t_off = obj->pwm->MATCHREL[(obj->pwm_ch * 2) + 0];
-    uint32_t t_on  = obj->pwm->MATCHREL[(obj->pwm_ch * 2) + 1];
+    uint32_t t_off = obj->pwm->MATCHREL[(obj->pwm_ch * 2) + 0] + 1;
+    uint32_t t_on  = obj->pwm->MATCHREL[(obj->pwm_ch * 2) + 1] + 1;
     float v = (float)t_on/(float)t_off;
-    obj->pwm->MATCHREL[(obj->pwm_ch * 2) + 0] = (uint32_t)us;
-    obj->pwm->MATCHREL[(obj->pwm_ch * 2) + 1] = (uint32_t)((float)us * (float)v);
+    obj->pwm->MATCHREL[(obj->pwm_ch * 2) + 0] = (uint32_t)us - 1;
+    if (us > 0) {
+        obj->pwm->MATCHREL[(obj->pwm_ch * 2) + 1] =  (uint32_t)((float)us * (float)v) - 1;
```

Comment:
would it be beneficial to add comments why us > 0 , what it handles, or lines like below `` obj->pwm->CTRL |= (1 << 2) | (1 << 3);  `` ? I without the reference manual have no clue what is going on. From the commit message, off by one errors are probably for lines 158 and 159, and then `` if (value != 1.0f) `` is probably for 100 percent error (also from the commit message).